pkg/tls/marshal.go

```diff
@@ -96,10 +101,30 @@ func ParseCert(b []byte) (*x509.Certificate, error) {
 			return x509.ParseCertificate(block.Bytes)
 		}
 	}
-
 	return nil, errors.New("failed to parse certificate")
```

Comment:
simplification: please can you make ParseCert() return ParseCertChain[0]? (like CertChainAsBytes but in reverse)